lib/src/internal/cdn_http_endpoints.dart

```diff
@@ -0,0 +1,216 @@
+import 'package:nyxx/src/core/snowflake.dart';
+import 'package:nyxx/src/internal/http/http_route.dart';
+import 'package:nyxx/src/internal/constants.dart';
+import 'package:nyxx/src/utils/utils.dart';
+
+abstract class ICdnHttpEndpoints {
+  String appAsset(Snowflake clientId, String assetHash, {String? format, int? size});
+
+  String appIcon(Snowflake clientId, String iconHash, {String? format, int? size});
+
+  String avatar(Snowflake id, String avatarHash, {String? format, int? size, bool animated = false});
+
+  String banner(Snowflake guildOrUserId, String hash, {String? format, int? size, bool animated = false});
+
+  String channelIcon(Snowflake channelId, String iconHash, {String? format, int? size});
+
+  String defaultAvatar(int discriminator, {int? size});
+
+  String emoji(Snowflake emojiId, {String? format, int? size});
+
+  String discoverySplash(Snowflake guildId, String splashHash, {String? format, int? size});
+
+  String memberAvatar(Snowflake guildId, Snowflake userId, String avatarHash, {String? format, int? size, bool animated = false});
+
+  String icon(Snowflake id, String iconHash, {String? format, int? size, bool animated = false});
+
+  String roleIcon(Snowflake roleId, String roleIconHash, {String? format, int? size});
+
+  String splash(Snowflake guildId, String splashHash, {String? format, int? size});
+
+  String sticker(Snowflake stickerId, {String? format = 'png'});
+
+  String stickerPackBanner(Snowflake bannerId, {String? format, int? size});
+
+  String teamIcon(Snowflake teamId, String teamIconHash, {String? format, int? size});
+
+  String guildEventCoverImage(Snowflake eventId, String eventCoverHash, {String? format, int? size});
+}
+
+class CdnHttpEndpoints implements ICdnHttpEndpoints {
+  String _makeAnimatableCdnUrl(ICdnHttpRoute fragment, String hash, {String? format = 'webp', int? size, bool animated = false}) {
+    if (hash.startsWith('a_') && animated) {
+      animated = true;
+    }
+
+    return _makeCdnUrl(fragment, format: format, size: size, animated: animated);
+  }
+
+  String _makeCdnUrl(ICdnHttpRoute fragments, {String? format = 'webp', int? size, bool animated = false}) {
+    format ??= 'webp';
```

Comment:
just remove nullable format parameter and ther wont be need for line 50

Reply:
Ok, I'm currently writing the docs for all the fields, so I'm not gonna do it now